src/MQTTClient.php

```diff
@@ -700,6 +700,12 @@ public function loop(bool $allowSleep = true, bool $exitWhenQueuesEmpty = false,
                     break;
                 }
             }
+
+            // Exit for real the loop after one round if limit -1
+            if ($queueWaitLimit == -1) {
+                $this->logger->debug('Force-exiting the main loop after one go');
+                break;
+            }
```

Comment:
I don't quite like this yet because I find it very non-intuitive. Can we use a different solution adding a new parameter (with a default)? One of these two solutions looks good to me:

*1) `$maxIterations` or `$exitAfterIterations`*
```php
 public function loop(
    bool $allowSleep = true, 
    bool $exitWhenQueuesEmpty = false, 
    int $queueWaitLimit = null,
    int $maxIterations = null
): void
{
    $iteration = 1;

    while (true) {
        // Do normal work

        if ($maxIterations !== null && $maxIterations <= $iteration) {
            break;
        }

        $iteration++;
    }
}
```

*2) `$runOnlyOnce`*
```php
 public function loop(
    bool $allowSleep = true, 
    bool $exitWhenQueuesEmpty = false, 
    int $queueWaitLimit = null,
    int $runOnlyOnce = false
): void
{
    while (true) {
        // Do normal work 

        if ($runOnlyOnce) {
            break;
        }
    }
}
```

The first solution seems more versatile, while the second one is extremely simple and the exit-logic could also be called before the other already existing exit-checks (like queue empty).

Reply:
I decided to revert the change as the `interrupt()` method inside an event loop callback actually does the job. No need to other another way to achieve the same result.